src/subscriptions.rs

```diff
@@ -1,20 +1,277 @@
-use bitcoin_hashes::sha256d::Hash as Sha256dHash;
+use bitcoin_hashes::{sha256d::Hash as Sha256dHash, Hash};
 use bitcoin_hashes::hex::FromHex;
+use bitcoin::blockdata::transaction::Transaction;
+use bitcoin::hash_types::{BlockHash, Txid};
 
 use rusoto_core::Region;
 use rusoto_dynamodb::{DynamoDb, DynamoDbClient, ScanInput};
+use rusoto_sqs::{ReceiveMessageRequest, Sqs, SqsClient, DeleteMessageRequest, GetQueueUrlRequest, GetQueueUrlResult, SendMessageRequest};
 
 use std::default::Default;
 use std::env;
 use std::format;
+use std::sync::{Arc};
+use std::sync::mpsc::{SyncSender};
+use std::collections::{HashMap, HashSet};
+use serde_json::Value;
+use std::time::Instant;
 
 use crate::errors::*;
-use std::collections::HashMap;
-use serde_json::Value;
+use crate::index::compute_script_hash;
+use crate::query::{Query};
+use crate::util::{spawn_thread, SyncChannel, HeaderEntry, FullHash};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+fn get_output_scripthash(txn: &Transaction, n: Option<usize>) -> Vec<FullHash> {
+    if let Some(out) = n {
+        vec![compute_script_hash(&txn.output[out].script_pubkey[..])]
+    } else {
+        txn.output
+            .iter()
+            .map(|o| compute_script_hash(&o.script_pubkey[..]))
+            .collect()
+    }
+}
+
+#[derive(Debug)]
+pub enum ScriptHashCompareMessage {
+    Start
+}
+
+pub struct ScriptHashComparer {
+    query: Arc<Query>,
+    pub chan: SyncChannel<ScriptHashCompareMessage>,
+    notifications_sender: SyncSender<SubscriptionMessage>,
+    in_progress: Arc<AtomicBool>
+}
+
+impl ScriptHashComparer {
+    pub fn new(query: Arc<Query>, notifications_sender: SyncSender<SubscriptionMessage>, in_progress: Arc<AtomicBool>) -> ScriptHashComparer {
+        ScriptHashComparer {
+            query,
+            chan: SyncChannel::new(3),
+            notifications_sender,
+            in_progress
+        }
+    }
+
+    fn compare_status_hashes(&mut self) {
+        info!("compare_status_hashes begin");
+        self.in_progress.store(true, Ordering::Relaxed);
+        let script_hashes_res = SubscriptionsManager::get_script_hashes();
+        if script_hashes_res.is_err() {
+            return ()
+        }
+        let script_hashes = script_hashes_res.unwrap();
+        info!("compare_status_hashes: script_hashes.len() = {}, starting", script_hashes.len());
+        let now = Instant::now();
+        for (i, (scripthash, old_statushash)) in script_hashes.iter().enumerate() {
+            if i % 1000 == 0 {
+                info!("Comparing scripthash #{}", i);
```

Comment:
Can we please keep the log of
`debug!("compare_status_hashes: comparing {} out of {}", i, script_hashes.len());` ?
I found it useful...